hudi-client/hudi-client-common/src/test/java/org/apache/hudi/metadata/TestHoodieBackedTableMetadataWriter.java

```diff
@@ -39,6 +55,160 @@
 import static org.mockito.Mockito.when;
 
 class TestHoodieBackedTableMetadataWriter {
+
+  private static Stream<Arguments> testArgumentsForEnabledPartitions() {
```

Comment:
These tests are moved from `TestMetadataPartitionType` and adjusted to consider table versions and whether the table is partitioned.